packages/utils/rbac.tsx

```diff
@@ -0,0 +1,113 @@
+import * as React from 'react';
+import {
+  AccessReviewResourceAttributes,
+  k8sCreate,
+  K8sVerb,
+} from '@openshift-console/dynamic-plugin-sdk';
+import * as _ from 'lodash-es';
+import log from 'loglevel';
+import { useSafetyFirst } from '../components/safety-first';
+import { ProjectModel, SelfSubjectAccessReviewModel } from '../models';
+import { SelfSubjectAccessReviewKind } from '../types';
+
+// Memoize the result so we only make the request once for each access review.
+// This does mean that the user will have to refresh the page to see updates.
+// Accept an `impersonateKey` parameter to include in the cache key even though
+// it's not used in the function body. (Impersonate headers are added
+// automatically by `k8sCreate`.) This function takes in the destructured
+// resource attributes so that the cache keys are stable. (`JSON.stringify` is
+// not guaranteed to give the same result for equivalent objects.)
+const checkAccessInternal = _.memoize(
+  (
+    group: string,
+    resource: string,
+    subresource: string,
+    verb: K8sVerb,
+    name: string,
+    namespace: string,
+    // eslint-disable-next-line no-unused-vars, @typescript-eslint/no-unused-vars
+    impersonateKey: string
```

Comment:
I don't see any use of this `impersonateKey` key at all. @alfonsomthd wdyt, can you think of any use case ??
else we can remove...